spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskService.java

```diff
@@ -139,15 +142,14 @@ public void executeTask(String taskName, Map<String, String> runtimeProperties,
 
 		taskDefinition = this.updateTaskProperties(taskDefinition);
 
-		Map<String, String> deploymentProperties = new HashMap<>(runtimeProperties);
-		URI uri = this.registry.find(String.format("task.%s",
-				taskDefinition.getRegisteredAppName()));
+		URI uri = this.registry.find(String.format("task.%s", taskDefinition.getRegisteredAppName()));
 		Resource resource = this.resourceLoader.getResource(uri.toString());
 
-		taskDefinition = qualifyProperties(taskDefinition, resource);
 
-		AppDeploymentRequest request = taskDefinition.createDeploymentRequest(resource,
-				deploymentProperties, runtimeParams);
+		Map<String, String> appDeploymentProperties = extractAppProperties(taskDefinition, taskDeploymentProperties);
+		Map<String, String> deployerDeploymentProperties = DeploymentPropertiesUtils.extractAndQualifyDeployerProperties(taskDeploymentProperties, taskDefinition.getName());
```

Comment:
To be even more correct, we want the label (even this does not commonly apply for tasks). But looking at the source, I think `TaskDefinition` should be fixed: one of the constructors does not call `super`, and I fear this can cause inconsistencies. See [here](https://github.com/spring-cloud/spring-cloud-dataflow/blob/28e62fd2b6c20202331c8448d75b3f4e50789345/spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/TaskDefinition.java#L40-L58)